src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java

```diff
@@ -249,16 +251,15 @@ public Result call() {
                                 try (final InputStreamReader inputStreamReader = new InputStreamReader(tempFilePath.read(), Charset.forName("UTF-8"))) {
                                     Gson gson = new GsonBuilder().create();
                                     Note n = gson.fromJson(inputStreamReader, Note.class);
-                                    JsonObject obj = gson.toJsonTree(n).getAsJsonObject();
-                                    JsonArray array = obj.get("paragraphs").getAsJsonArray();
-                                    for (JsonElement element : array)
-                                        if (element.isJsonObject()) {
-                                            // get each cell form the JSON element
-                                            JsonObject cell = element.getAsJsonObject();
-                                            String code = cell.get("text").getAsString();
-                                            listener.getLogger().println(code);
-                                            listener.getLogger().println(interpreterManager.invokeInterpreter(code, task, ws));
+                                    for (Paragraph para : n.getParagraphs()) {
+                                        // skipping markdowns
+                                        if (para.getConfig().get("editorMode").equals("ace/mode/markdown")) {
```

Comment:
Ditto here, might be useful to explain where `"ace/mode/markdown"` comes from. Maybe add a constant somewhere and use in both classes.